### ISSUE TITLE
if user Change the Download Folder From AppData to any folder inside Plursight Offline App

### ISSUE THREAD

**Issue by MaxBayne** [KevinWang15/PsVDecrypt#5]
this app work fine when Plursight Offline App store Downloaded Videos inside AppData so he go to search inside this folder to get the courses that downloaded by Plursight Application so what if use go to Perference inside PlursightApp And change the Download Location so PSVDecrypt fail to catch the courses so need to check this issue ... thanks

**Comment by samdx** [comment]
This line, I think:

```csharp
var coursesdir = Path.Combine(
                Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
                "Pluralsight", "courses");
```
The `LocalApplicationData` should be changed to `args[0]` to specific a folder, but the database file will still remain the same place.

**Comment by KevinWang15** [comment]
This tool only covers the most common use case as I don't really have much time to make it very adaptive.
It's open source so you are welcome to modify the code to suit your needs (like you would to a bash script) 😃
As has been pointed out, in this case, you only need to modify the ```coursesdir``` variable.

**Comment by samdx** [comment]
Hi,

I have just created a simple GUI for those who has the same demand.
- Simpe, Ugly, but works

Download [here](https://transfer.sh/1PePQ/PsVDecrypt-1.1.0.0.zip): [https://transfer.sh/1PePQ/PsVDecrypt-1.1.0.0.zip](https://transfer.sh/1PePQ/PsVDecrypt-1.1.0.0.zip)

---

The first time I came to the C# world. I don't have any background with these things, and these made me confused a lot, either .NET, .NET Core,..
And man, Visual Studio is really slow as...

**Comment by KevinWang15** [comment]
@samdx Cool! Thanks for your interest in making it better.
I will definitely check it out and fix some other issues in the coming months.

**Comment by samdx** [comment]
Yeah, mate

I'm not good at code so I don't want to create a PR (yet).

I had created a separated branch in my fork at samdx/PsVDecrypt@4c5e96a if you interested.

The issue now is the app hangs when a child thead is running. I will come back with this issue later.

**Comment by kousherAlam** [comment]
@samdx Put project on github bro

**Comment by samdx** [comment]
@kousherAlam Yes, of course. Please check my fork given above for further commits. My code is 'a bit ugly' so I don't want to create a PR right now. 😸

**Comment by isitmeor** [comment]
how do i change the paths in the windows version?

**Comment by marcosd87** [comment]
@samdx i've tryied to download but it says "not found" :(

**Comment by samdx** [comment]
@marcosd87 Have you selected the input folder which contains the `courses` folder and set the output folder?

Provide more info, please. How you did, what you got, etc.

**Comment by cpt-kernel** [comment]
@samdx The GUI hangs (Windows10) when I click 'decrypt'. Version 1.0.0 does work though. 
Anything I can do to help just let me know.

**Comment by azharbf** [comment]
@samdx 

please can you help me convert the psv to mp4. i downloaded the PsVDecrypt-1.1.0.0.zip, but not able to convert. your help will be highly appreciated.

**Comment by KevinWang15** [comment]
@azharbf 
What difficulties are you encountering exactly?
Just tried with the latest Offline Player and it still works (on Win 10).

Sign in with Offline Player，
Make sure in the Preferences, Download Location is C:\Users\<username>\AppData\Local\Pluralsight
Search and download courses,
wait for all downloads to finish.
Then unzip everything from https://github.com/KevinWang15/PsVDecrypt/releases/download/1.0.0/Release-1.0.0.zip and run PsVDecrypt.exe

It should show:

```
Courses directory: C:\Users\<username>\AppData\Local\Pluralsight\courses
Output Directory: D:\PsV\output

Found 3 course(s):
 > xxx-xxx
 > yyy-yyyy
 > zzz-zzzzz

Press any key to start decrypting all courses..
```

Press any key and the process will start. Wait for it to finish, and decrypted .mp4 files are found at ./output

**Comment by Kang-Jack** [comment]
submitted a pull request to resolve this
